source/common/network/address_impl.cc

```diff
@@ -161,7 +161,8 @@ IoHandlePtr InstanceBase::socketFromSocketType(SocketType socketType) const {
   }
 
   IoHandlePtr io_handle = std::make_unique<IoSocketHandle>(::socket(domain, flags, 0));
-  RELEASE_ASSERT(io_handle->fd() != -1, "");
+  RELEASE_ASSERT(io_handle->fd() != -1,
+                 fmt::format("socket(2) failed, got error: {}", strerror(errno)));
```

Comment:
I think this might be an unsafe use of `errno`, since `io_handle->fd()` might mutate? The root problem here is the direct use of `::socket()` above, which should ideally be going through https://github.com/envoyproxy/envoy/blob/master/include/envoy/api/os_sys_calls.h#L126 to safely handle errno and mocking.

Reply:
Done. Switched it to using socket from api/os_sys_calls.